universal-application-tool-0.0.1/app/views/AzureFileUploadViewStrategy.java

```diff
@@ -0,0 +1,220 @@
+package views;
+
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.footer;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.text;
+import static j2html.attributes.Attr.FORM;
+import static views.BaseHtmlView.makeCsrfTokenInputTag;
+import static views.BaseHtmlView.submitButton;
+
+import controllers.applicant.routes;
+import j2html.TagCreator;
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import java.util.Optional;
+import javax.inject.Inject;
+import services.MessageKey;
+import services.applicant.question.FileUploadQuestion;
+import services.cloud.FileNameFormatter;
+import services.cloud.StorageUploadRequest;
+import services.cloud.azure.BlobStorageUploadRequest;
+import views.applicant.ApplicantProgramBlockEditView.Params;
+import views.questiontypes.ApplicantQuestionRendererFactory;
+import views.questiontypes.ApplicantQuestionRendererParams;
+import views.style.ApplicantStyles;
+import views.style.Styles;
+
+public class AzureFileUploadViewStrategy extends FileUploadViewStrategy {
```

Comment:
suggestion (blocking): it looks like a lot of the markup code in here was copy-pasted from `AwsFileUploadViewStrategy`. Please reduce the duplication between the two classes.

Reply:
Done.